Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -1054,6 +1063,83 @@ void ABenchmarkGymGameModeBase::CheckVelocityForPlayerMovement()
 	}
 }
 
+void ABenchmarkGymGameModeBase::OutputPlayerDensity()
+{
+	// Outputs the count that each NPC and Simulated Player falls into each of the QBI-F bucket types. This is not performant but
+	// is only used for debugging purposes currently and isn't enabled by default.
+	FTimerHandle CountTimer;
+	GetWorld()->GetTimerManager().SetTimer(
+		CountTimer,
+		[WeakThis = TWeakObjectPtr<ABenchmarkGymGameModeBase>(this)]() {
+		if (ABenchmarkGymGameModeBase* GameMode = WeakThis.Get())
+		{
+			USpatialNetDriver* SpatialDriver = Cast<USpatialNetDriver>(GameMode->GetNetDriver());
+
+			TArray<AActor*> PlayerControllers, PlayerCharacters, NPCs, AllCharacters;
+			UGameplayStatics::GetAllActorsOfClass(GameMode->GetWorld(), GameMode->SimulatedPlayerControllerClass, PlayerControllers);
+			UGameplayStatics::GetAllActorsOfClass(GameMode->GetWorld(), GameMode->SimulatedPawnClass, PlayerCharacters);
+			UGameplayStatics::GetAllActorsOfClass(GameMode->GetWorld(), GameMode->NPCClass, NPCs);
+			AllCharacters = PlayerCharacters;
+			AllCharacters.Append(NPCs);
+
+			const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
+			TArray<float> NCDDistanceRatios;
+			NCDDistanceRatios.Push(SpatialGDKSettings->FullFrequencyNetCullDistanceRatio);
+			for (auto i : SpatialGDKSettings->InterestRangeFrequencyPairs)
+			{
+				NCDDistanceRatios.Push(i.DistanceRatio);
+			}
+			NCDDistanceRatios.Push(MAX_FLT);
```

Comment:
Does this break the full frequency bucket into two?

Reply:
Will add a comment - this is to catch actors that are outside a client's interest range